poly-commitment/src/srs.rs

```diff
@@ -265,6 +266,16 @@ impl<G: CommitmentCurve> SRS<G> {
             lagrange_bases: HashMap::new(),
         }
     }
+
+    /// Absorb all the information of the verification key into the given sponge.
+    ///
+    /// It is useful in SNARK protocols where the verification key must be
+    /// absorbed at the beginning of the protocol.
+    /// It does absorb first the basis G and after that the blinder H.
+    pub fn digest<Sponge: FqSponge<G::BaseField, G, G::ScalarField>>(&self, sponge: &mut Sponge) {
+        self.g.iter().for_each(|g| sponge.absorb_g(&[*g]));
```

Comment:
Remind me, how are `g` generated? They're not independent, right? If they're not (e.g. they're powers of each other, or progressive hashes), maybe we can hash just first/last value...

Reply:
They can be arbitrary values for the IPA.